broker-node/app/Clients/BrokerNode.php

```diff
@@ -380,25 +375,24 @@ public static function buildTransactionData($chunks)
         return $request;
     }
 
-    public static function getTransactionsToApprove(&$request)
+    private static function getTransactionsToApprove(&$request)
     {
         self::initIri();
 
         $command = new \stdClass();
         $command->command = "getTransactionsToApprove";
         $command->depth = IriData::$depthToSearchForTxs;
 
-        BrokerNode::$iriRequestInProgress = true;
 
         $result = self::$IriWrapper->makeRequest($command);
 
-        BrokerNode::$iriRequestInProgress = false;
 
         if (!is_null($result) && property_exists($result, 'branchTransaction')) {
             //switching trunk and branch
             //do we do this randomly or every time?
+            $tmp_trunk = $request->trunkTransaction;
             $request->trunkTransaction = $result->branchTransaction;
-            $request->branchTransaction = $result->trunkTransaction;
```

Comment:
@rfornea I think this was confirming the branch as both branch and trunk? Not sure if these might have caused any slowdowns or problems?

Reply:
It just swaps them.  Something about entropy.